rosapi/src/rosapi/params.py

```diff
@@ -42,43 +43,82 @@
 # rospy parameter server isn't thread-safe
 param_server_lock = threading.RLock()
 
-def set_param(name, value):
-    d = None
-    try:
-        d = loads(value)
-    except ValueError:
-        raise Exception("Due to the type flexibility of the ROS parameter server, the value argument to set_param must be a JSON-formatted string.")
-    with param_server_lock:
-        rospy.set_param(name, d)
-    
-    
-def get_param(name, default):
-    d = None
-    if default is not "":
+def set_param(name, value, params_glob):
+    if params_glob and not any(fnmatch.fnmatch(str(name), glob) for glob in params_glob):
+        # If the glob list is not empty and there are no glob matches,
+        # stop the attempt to set the parameter.
+        return
+    else:
```

Comment:
You can drop the else (and subsequent indentation here, because you return in the if's body above. That should help Github see that you didn't actually change all the lines in the else part :)